### DISCUSSION TITLE
Strange behavior of `Ref`s

### DISCUSSION THREAD

**Discussion by ndonkoHenri** [flet-dev/flet#506] [Q&A] [answered]
Refs have a strange behavior. I dont know if this is a bug or its normal/intentional..
I used a ref[textfield] to update a button's text.

```python
from flet import *

def main(page: Page):
    page.title = "Ref Misbehavior"
    page.window_height = 200
    page.window_width = 400

    def click(e):
        # TextField has no property text, but FilledButton.text is successfully updated
        fake_ref.current.text = "Hello France"
        page.update()

    fake_ref = Ref[TextField]()  # changing from Ref[TextField] to Ref[Text] still works

    page.add(
        FilledButton(
            "Hello Estonia",
            fake_ref,
            on_click=click
        )
    )

flet.app(target=main,)
```
 **Can someone please explain why this works as such??**

using `stack` or `Text` instead of `textfield` still works
surely any control
I expected it to return an error, RefError..
telling me an invalid Ref was used

**Comment by FeodorFitsner** [comment]
TextField doesn't have "text" properly, but it has "value".

**Reply by ndonkoHenri** [reply]
You maybe didn't understand my question.
So, I have a `Button` and I passed a `Ref[Textfield]` to it's ref attribute.
And from there I tried to update the `text` property of that `Button`.

The thing is, how/why can `Ref[Textfield]` be used to update properties of a Button????
`fake_ref.current` normally has no attribute `text`, since it's kind of a `Textfield`, but successfully updated the button.

**Comment by FeodorFitsner** [comment, marked as answer]
Because in Python `Ref[Textfield]` is a generic of type `Textfield`. It's a type "hint". Python does not enforce types - you can do whatever you like.

**Reply by ndonkoHenri** [reply]
Ah i see. Thanks. ✅
